.github/workflows/release.yml

```diff
@@ -20,34 +31,69 @@ jobs:
       - name: Build and Pack
         run: dotnet pack
           --configuration Release
-          --output "${{github.workspace}}/artifacts/packages"
+          --output "${{ github.workspace }}/artifacts/packages"
           /p:PackageProjectUrl="${{ github.server_url }}/${{ github.repository }}/tree/${{ github.event.release.tag_name }}"
           /p:PackageReleaseNotes="${{ github.server_url }}/${{ github.repository }}/blob/${{ github.event.release.tag_name }}/CHANGELOG.md"
+          ${{ env.version_suffix_args }}
 
-      - name: Test
+      - name: Unit Test
+        run: dotnet test
+          --configuration Release
+          --filter="TestCategory=Smoke&TestCategory!=Manual"
+          --logger "trx;LogFileName=${{ github.workspace }}/artifacts/test-results/smoke.trx"
+          ${{ env.version_suffix_args }}
+          
+      - name: Run Live Tests
         run: dotnet test
           --configuration Release
-          --filter="TestCategory~Online"
-          --logger "trx;LogFileName=${{github.workspace}}/artifacts/test-results/full.trx"
+          --filter="TestCategory!=Smoke&TestCategory!=Images&TestCategory!=Moderations&TestCategory!=Manual"
+          --logger "trx;LogFilePrefix=live"
+          --results-directory ${{ github.workspace }}/artifacts/test-results
+          ${{ env.version_suffix_args }}
         env:
           OPENAI_API_KEY: ${{ secrets.OPENAI_API_KEY }}
 
+      - name: Upload artifact
+        uses: actions/upload-artifact@v2
+        if: ${{ !cancelled() }}
+        with:
+          name: build-artifacts
+          path: ${{ github.workspace }}/artifacts
+
+  deploy:
+    name: Publish Package
+    needs: build
+    if: github.event_name != 'pull_request'
```

Comment:
is it even possible for this to be triggered from a PR?

Reply:
condition removed